src/indieweb_utils/webmentions/validate.py

```diff
@@ -0,0 +1,157 @@
+from urllib import parse as url_parse
+from typing import List
+
+from bs4 import BeautifulSoup
+import requests
+
+from ..utils.urls import canonicalize_url
+
+
+class WebmentionValidationError(Exception):
+    pass
+
+
+class WebmentionIsGone(Exception):
+    pass
+
+
+def process_vouch(vouch: str, source: str, vouch_list: List[str]) -> bool:
+    # use vouch to flag webmentions for moderation
+    # see Vouch spec for more: https://indieweb.org/Vouch
+    # by default, webmention should require moderation
+    # if a vouch is valid, webmention does not need moderation
+
+    moderate = True
+
+    if vouch and vouch != "":
+        vouch_domain = url_parse.urlparse(vouch).netloc
+
+        if moderate:
+            # only get domains
+            vouch_list = [v[0] for v in vouch_list]
```

Comment:
The vouch_list is a bit of a mystery to me. The type indicates that it's a list of strings and as such I'd expect `v[0]` to be a single character, but the comment says the first element is a domain? Should update the docstrings / typehint to match the type.

Reply:
I have removed the line of code in question. Good catch! I believe it was part of a legacy implementation that I used on my webmention endpoint.

Overall, the function implements the vouch extension to the webmention specification. Using this function, we can evaluate whether a webmention is likely to need moderation. If a vouch passes, we say a webmention does not need moderation.